### PR TITLE
Quicktest: check snapshot VDI fields

### PR DESCRIPTION
**Pull request by gaborigloi**
And reorganize the helpers a bit.

This test has already found an issue, CA-290241, and it's passing on local ext and NFS.

Signed-off-by: Gabor Igloi <gabor.igloi@citrix.com>

### CODE REVIEW

File: ocaml/quicktest/quicktest_common.ml [hunk 1]
```diff
@@ -117,7 +98,23 @@ let compare_fields cls fields original_rec new_rec =
...
-          "%s field %s should should be different, but is the same: '%s'"
+          "%s field %s should be different, but is the same: '%s'"
           cls field_name new_field
   in
   List.iter check fields
+
+module Time = struct
+  type t = float
+
+  let now () = Unix.gettimeofday ()
```
Comment (mseri) [inline ocaml/quicktest/quicktest_common.ml]: I think you also have Mtime at your disposal if you want something even more reliable

File: ocaml/quicktest/quicktest_common.ml [hunk 2]
```diff
@@ -64,35 +48,32 @@ let cli_cmd args =
...
-  match List.filter (fun (_, r) -> List.mem_assoc Xapi_globs.is_guest_installer_network r.API.network_other_config) all with
-  | (rf, _) :: _ -> rf
-  | _ -> failwith "Could not find guest installer network"
-
-(** Return a host's domain zero *)
-let dom0_of_host session_id host =
-  Client.Host.get_control_domain !rpc session_id host
-
-let get_default_sr session_id =
-  Client.Pool.get_default_SR ~session_id ~rpc:!rpc ~self:(get_pool session_id)
```
Comment (mseri) [inline ocaml/quicktest/quicktest_common.ml]: So this was unused!

File: ocaml/quicktest/quicktest_http.ml [hunk 3]
```diff
@@ -59,14 +59,6 @@ end
...
 module Secret_Auth_fails = struct (* {{{1 *)
 
-  let http request f =
-    let open Xmlrpc_client in
-    let transport =
-      if !Quicktest_args.using_unix_domain_socket
-      then Unix Xapi_globs.unix_domain_socket
-      else SSL(SSL.make ~use_fork_exec_helper:false (), !Quicktest_args.host, 443) in
-    with_transport transport (with_http request f)
-
```
Comment (minishrink) [inline ocaml/quicktest/quicktest_http.ml]: I take it this is obsolete?

Comment (gaborigloi) [reply ocaml/quicktest/quicktest_http.ml]: No, it was duplicated in quicktest_common

File: ocaml/quicktest/quicktest_common.ml (hunk 1 above)
Comment (gaborigloi) [reply ocaml/quicktest/quicktest_common.ml]: I used stdext-date because that type is stored in the VDI record

**Pull request merged**